### PR TITLE
Extract the suspicious files from the traces : Closes #1623

### LINKED ISSUE #1623: Extract the suspicious files from the traces
Motivation
----------
Currently the suspicious files are reported by the conveyor, the pilot and `rucio download`. We can centralize everything so that Kronos checks the traces and reports as suspicious all the files with errors that match a certain pattern

### PR DESCRIPTION
**Pull request by cserf**
Extract the suspicious files from the traces : Closes #1623

### CODE REVIEW

File: lib/rucio/daemons/tracer/kronos.py [hunk 1]
```diff
@@ -127,6 +138,17 @@ def __update_atime(self):
         rses = []
         for report in self.__reports:
             try:
+                # Identify suspicious files
+                if self.__bad_files_patterns and report['eventType'] in ('get_sm', 'sm_get', 'get_sm_a', 'sm_get_a') and report['clientState'].find('FAILED') > -1:
```
Comment (tbeerman) [inline lib/rucio/daemons/tracer/kronos.py]: You can remove `sm_get` and `sm_get_a`. They were traces sent with eventType some time ago by mistake but it is fixed now and they should not appear anymore.

Comment (tbeerman) [inline lib/rucio/daemons/tracer/kronos.py]: Are you sure that get all needed clientStates with this? Looking at the last 30 days only at `get_sm/_a` I can see a lot of different error states and they do not all have "FAILED" in the name. Here is a list of all clientStates in the last 30 days that are not `DONE`, `FOUND_ROOT` or `ALREADY_DONE` ordered by count.

- `STAGEIN_ATTEMPT_FAILED`
- `NO_REPLICA`
- `STAGEIN_FILE_FAILED`
- `COPY_ERROR`
- `STAGEIN_FAILED`
- `NO_FILE`
- `CP_TIMEOUT`
- `GLOBUS_FAIL`
- `AD_MISMATCH`
- `BAD_COPYTOOL`
- `STAGEIN_BAD_COPYTOOL`
- `NO_SPACE`
- `NO_PROTOCOL_DEFINED`
- `ServiceUnavailable`
- `FILE_EXISTS`

File: lib/rucio/daemons/tracer/kronos.py [hunk 2]
```diff
@@ -250,11 +272,20 @@ def kronos_file(once=False, thread=0, brokers_resolved=None, dataset_queue=None)
     chunksize = config_get_int('tracer-kronos', 'chunksize')
     prefetch_size = config_get_int('tracer-kronos', 'prefetch_size')
     subscription_id = config_get('tracer-kronos', 'subscription_id')
+    try:
+        bad_files_patterns = []
+        pattern = get(section='kronos', option='bad_files_patterns', session=None)
```
Comment (tbeerman) [inline lib/rucio/daemons/tracer/kronos.py:277]: why not just use `config_get`?

**Review by tbeerman — DISMISSED** [review]

File: lib/rucio/daemons/tracer/kronos.py (hunk 2 above)
Comment (cserf) [reply lib/rucio/daemons/tracer/kronos.py:277]: `config_get` gets the config from the `rucio.cfg`. It's better to get it from the config table.

File: lib/rucio/daemons/tracer/kronos.py (hunk 1 above)
Comment (cserf) [reply lib/rucio/daemons/tracer/kronos.py]: Where all the states are documented ?

Comment (cserf) [reply lib/rucio/daemons/tracer/kronos.py]: OK

**Review by bari12 — APPROVED** [review]

**Pull request merged**